content/docs/idp/well-architected/patterns/security-updates-using-components.md

```diff
@@ -0,0 +1,204 @@
+      instanceClass: args.instanceClass,
+      allocatedStorage: args.storage,
+
+      // Security defaults managed by platform team
+      storageEncrypted: true,
+      kmsKeyId: args.kmsKeyId,
+      dbSubnetGroupName: subnetGroup.name,
+      vpcSecurityGroupIds: [this.createSecurityGroup(args.vpcId)],
+
+      // Updated security settings (v1.2.3)
+      enabledCloudwatchLogsExports: ["postgresql", "upgrade"],
+      monitoringInterval: 60,
+      performanceInsightsEnabled: true,
+
+      // Force SSL connections
+      parameterGroupName: this.createParameterGroup(),
+
+      // Automated backups with encryption
+      backupRetentionPeriod: args.backupRetentionDays,
+      backupWindow: "03:00-04:00",
+      maintenanceWindow: "sun:04:00-sun:05:00",
+
+      tags: {
+        SecurityCompliant: "true",
+        ComponentVersion: "1.2.3",
+        LastSecurityUpdate: "2024-01-15",
+      },
+    }, { parent: this });
+  }
+
+  private createSecurityGroup(vpcId: string): aws.ec2.SecurityGroup {
+    return new aws.ec2.SecurityGroup("db-sg", {
+      vpcId: vpcId,
+      ingress: [{
+        fromPort: 5432,
+        toPort: 5432,
+        protocol: "tcp",
+        cidrBlocks: ["10.0.0.0/8"], // Only internal traffic
+      }],
+      // No egress rules - deny all outbound
+    }, { parent: this });
+  }
+
+  private createParameterGroup(): aws.rds.ParameterGroup {
+    return new aws.rds.ParameterGroup("secure-pg", {
+      family: "postgres14",
+      parameters: [
+        { name: "ssl", value: "on" },
+        { name: "log_statement", value: "all" },
+        { name: "log_min_duration_statement", value: "1000" },
+      ],
+    }, { parent: this });
+  }
+}
+```
+
+Platform team creates policy to enforce current component versions:
+
+```typescript
+// policies/component-version-policy.ts
+import { PolicyPack, validateResourceOfType } from "@pulumi/policy";
+import { aws } from "@pulumi/aws";
+
+const MINIMUM_COMPONENT_VERSIONS = {
+  "SecureDatabase": "1.2.0",
+  "SecureNetwork": "2.1.0",
+  "SecureStorage": "1.5.0",
+};
+
+function isVersionOutdated(currentVersion: string, minimumVersion: string): boolean {
+  const current = currentVersion.split('.').map(n => parseInt(n));
+  const minimum = minimumVersion.split('.').map(n => parseInt(n));
+
+  for (let i = 0; i < Math.max(current.length, minimum.length); i++) {
+    const currentPart = current[i] || 0;
+    const minimumPart = minimum[i] || 0;
+
+    if (currentPart < minimumPart) return true;
+    if (currentPart > minimumPart) return false;
+  }
+  return false;
+}
+
+new PolicyPack("component-version-enforcement", {
+  policies: [{
+    name: "enforce-minimum-component-versions",
+    description: "Ensure all components are using minimum required versions for security",
+    enforcementLevel: "mandatory",
+    validateResource: validateResourceOfType(aws.rds.Instance, (instance, args, reportViolation) => {
+      const componentVersion = instance.tags?.ComponentVersion;
```

Comment:
This indicates to me that we need some metadata for policies that specifically apply to components, or a cleaner way to implement this use case.